content/en/blog/github-swiss-knife-3/index.md

```diff
@@ -0,0 +1,203 @@
+
+There have also been some problems regarding the notifications. The Dependabot integration
+creates a lot of pull requests for several repositories, and sometimes,
+the information flood is too much. Therefore, we have custom labeling functionality
+that helps us automatically label PRs based on the files the changes affect.
+The automatic labeling is implemented with [GitHub Action](https://github.com/actions/labeler),
+which is triggered on PR creation.
+We can filter only the PR notifications needed for our chat integration using these labels.
+
+Automatic labeling can be tremendously helpful, especially with large repositories
+with many subprojects. Labels can also be used as a filter basis for GitHub Actions.
+For instance, one can configure a specific test for changes that affect only a particular folder.
+This approach can speed up the CI runtimes and save resources.
+
+## Codespaces And CLI Tool
+
+[GitHub Codespaces](https://github.com/features/codespaces) was a less-used feature for our project.
+Still, [we found it helpful](https://github.com/findy-network/agency-workshop-codespace),
+especially when we organized workshops to help developers get started with agency application development.
+
+Codespaces offers a cloud-based development environment that works as
+[VS Code dev containers](https://code.visualstudio.com/docs/devcontainers/containers).
+A repository can have a configuration for the environment, and the codespace is created and
+launched based on that configuration. Once the codespace is up and running, the developers have
+a ready development environment with all the needed bells and whistles. It removes a lot of
+the setup hassle we usually encounter when using different operating systems and tools for development.
+
+*Example of [a codespace configuration](https://github.com/findy-network/agency-workshop-codespace/blob/master/.devcontainer/devcontainer.json)
+for our workshop participants. All the needed tools are preinstalled
+to the codespace that one may need when doing the workshop exercises:*
+
+```json
+{
+  "name": "Findy Agency workshop environment",
+  "image": "ghcr.io/findy-network/agency-workshop-codespace:0.1.0",
+  "workspaceFolder": "/home/vscode/workshop",
+  "features": {
+    "ghcr.io/devcontainers-contrib/features/direnv-asdf:2": {}
+  },
+  "customizations": {
+    "vscode": {
+      "extensions": [
+        "golang.go",
+        "mathiasfrohlich.Kotlin"
+      ]
+    }
+  },
+  "postCreateCommand": "asdf direnv setup --shell bash --version latest"
+}
+```
+
+[The GitHub CLI](https://cli.github.com/) tool was another helpful tool we found during the agency project.
+It offers the GitHub web application functionality via the command line. The tool is especially useful
+for those who suffer from context switching
+from terminal to browser and vice versa or who prefer to work from a terminal altogether.
+In addition, the CLI tool can be used to automate certain routines, specifically in CI.
```

Comment:
Other cons of `gh` are that it provides a convenient playground for GitHub APIs and lets you play with JSON payloads quickly. All of that provides an excellent platform to build IDE/PDE add-ons. 

In `find-network` projects, the `Makefile` rule implementation has been a great benefit -- personally, at least.